sql/parser/builtins.go

```diff
@@ -943,6 +961,89 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+func regexpReplace(s, pattern, to, flags string) (Datum, error) {
+	pattern, global := regexpEvalFlags(pattern, flags)
+
+	patternRe, err := regexp.Compile(pattern)
+	if err != nil {
+		return DNull, err
+	}
+
+	matchCount := 1
+	if global {
+		matchCount = -1
+	}
+
+	replaceIndex := 0
+	var newString bytes.Buffer
+	for _, matchIndex := range patternRe.FindAllStringSubmatchIndex(s, matchCount) {
```

Comment:
this might be more natural with https://godoc.org/regexp#Regexp.ReplaceAllStringFunc